### PR TITLE
Change surface_azimuth convention in get_pvgis_hourly

### LINKED ISSUE #1724: `pvlib.iotools.get_pvgis_hourly`'s `surface_azimuth` parameter doesn't use pvlib's azimuth convention
Nearly everything in pvlib represents azimuth angles as values in [0, 360) clockwise from north, except `pvlib.iotools.get_pvgis_hourly`:

https://github.com/pvlib/pvlib-python/blob/3def7e3375002ee3a5492b7bc609d3fb63a8edb1/pvlib/iotools/pvgis.py#L79-L81

This inconsistency is a shame.  However, I don't see any way to switch it to pvlib's convention without a hard break, which is also a shame.  I wonder how others view the cost/benefit analysis here.

See also https://github.com/pvlib/pvlib-python/pull/1395#discussion_r1181853794

### PR DESCRIPTION
**Pull request by AdamRJensen**
- [x] Closes #1724
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - ~~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

Nearly everything in pvlib represents azimuth angles as values in [0, 360) clockwise from north, except ``pvlib.iotools.get_pvgis_hourly``. This PR modifies the get_pvgis_hourly function such that it follows the pvlib convention.

### CODE REVIEW

File: pvlib/iotools/pvgis.py [hunk 1]
```diff
@@ -76,9 +76,10 @@ def get_pvgis_hourly(latitude, longitude, start=None, end=None,
         Otherwise only global irradiance is returned.
     surface_tilt: float, default: 0
         Tilt angle from horizontal plane. Ignored for two-axis tracking.
-    surface_azimuth: float, default: 0
-        Orientation (azimuth angle) of the (fixed) plane. 0=south, 90=west,
-        -90: east. Ignored for tracking systems.
+    surface_azimuth: float, default: 180
+        Orientation (azimuth angle) of the (fixed) plane. Counter-clockwise
+        from north (north=0, south=180). This is offset 180 degrees from
+        the convention used by PVGIS. Ignored for tracking systems.
```
Comment (kandersolar) [inline pvlib/iotools/pvgis.py:82]: Perhaps a `.. versionchanged::` is warranted in this parameter description?

File: docs/sphinx/source/whatsnew/v0.9.6.rst [hunk 2]
```diff
@@ -5,6 +5,13 @@ v0.9.6 (Anticipated June 2023)
 ------------------------------
 
 
+Breaking Changes
+~~~~~~~~~~~~~~~~
+* Modified the ``surface_azimuth`` parameter in :py:func:`pvlib.iotools.get_pvigs_hourly` to conform to the
```
Comment (kandersolar) [inline docs/sphinx/source/whatsnew/v0.9.6.rst:10]:
```suggestion
* Modified the ``surface_azimuth`` parameter in :py:func:`pvlib.iotools.get_pvgis_hourly` to conform to the
```

File: pvlib/iotools/pvgis.py [hunk 3]
```diff
@@ -136,6 +137,12 @@ def get_pvgis_hourly(latitude, longitude, start=None, end=None,
         the error message in the response will be raised as an exception,
         otherwise raise whatever ``HTTP/1.1`` error occurred
 
+    Info
+    ----
```
Comment (kandersolar) [inline pvlib/iotools/pvgis.py:141]:
```suggestion
    Notes
    -----
```
https://numpydoc.readthedocs.io/en/latest/format.html#notes

**Review by kandersolar — COMMENTED** [review]
The supplied azimuth is echoed in the `inputs` return value... I guess we should translate that one as well?  Or maybe just get rid of that return value altogether so that the function can have the same `data, metadata` return structure as the rest of iotools?

**Comment by AdamRJensen** [comment]
> The supplied azimuth is echoed in the `inputs` return value... I guess we should translate that one as well? Or maybe just get rid of that return value altogether so that the function can have the same `data, metadata` return structure as the rest of iotools?

So the second returned element "inputs" actually also contains metadata describing the outputs, so it is useful. I am a strong believer in sticking with the (data, meta) output format and think that the output related content in "inputs" should simply be included in the meta dictionary and the input related content in "inputs" should be discarded. Consistency is key, and I always forget how many items the PVGIS functions return. But changing this would be a breaking change...?

**Review by kandersolar — COMMENTED** [review]
Regarding `inputs`: how about we translate azimuth in this PR and open a new issue for changing the number of return values?  Might want to get that into 0.10.0 too.

**Comment by AdamRJensen** [comment]
> Regarding `inputs`: how about we translate azimuth in this PR and open a new issue for changing the number of return values? Might want to get that into 0.10.0 too.

Translating the azimuth is surprisingly convoluted. First off, it is necessary to update both the description: 

    metadata['inputs']['mounting_system']['fields']['azimuth']['description'] = \
        'Orientation (azimuth) angle of the (fixed) PV system (0 = N, 90 = E, 180 = S)'
and the value (changing the value requires some wrangling due to the field name changing depending on the type of mounting system):

    inputs['mounting_system'][list(inputs['mounting_system'].keys())[0]]['azimuth']['value'] = \
        inputs['mounting_system'][list(inputs['mounting_system'].keys())[0]]['azimuth']['value'] + 180

However, the above line is not sufficient, given that for certain tracking mounting systems the azimuth value is specified as '-'.

Also, the metadata structure is completely different for the json and the csv output formats. In short, translating the azimuth will add significant lines of code to the function, which I don't think is worth it. I'm more inclined to add a warning message stating that the metadata concerning the azimuth is not corrected. @kandersolar what do you think of this?

File: pvlib/iotools/pvgis.py [hunk 4]
```diff
@@ -76,9 +76,15 @@ def get_pvgis_hourly(latitude, longitude, start=None, end=None,
...
-        -90: east. Ignored for tracking systems.
+    surface_azimuth: float, default: 180
+        Orientation (azimuth angle) of the (fixed) plane. Counter-clockwise
+        from north (north=0, south=180). This is offset 180 degrees from
+        the convention used by PVGIS. Ignored for tracking systems.
+
+        .. versionchanged:: 0.10.0
+           The `surface_azimuth` parameter now follows the pvlib convention, which
+           is counterclockwise from north. However, the convention used by the
+           PVGIS website and pvlib<=0.9.6 is offset by 180 degrees.
```
Comment (kandersolar) [inline pvlib/iotools/pvgis.py:87]:
```suggestion
           PVGIS website and pvlib<=0.9.5 is offset by 180 degrees.
```

**Review by kandersolar — COMMENTED** [review]
> I'm more inclined to add a warning message stating that the metadata concerning the azimuth is not corrected. @kandersolar what do you think of this?

I guess a "warning" does not refer to `warnings.warn()` but rather text in the docstring?  If so, ok with me!

**Review by kandersolar — APPROVED** [review]
LGTM!

**Pull request merged**